crates/forge/src/cmd/test/mod.rs

```diff
@@ -312,18 +313,54 @@ impl TestArgs {
 
         let sources_to_compile = self.get_sources_to_compile(&config, &filter)?;
 
-        let mut compiler = ProjectCompiler::new()
-            .dynamic_test_linking(config.dynamic_test_linking)
-            .quiet(shell::is_json() || self.junit)
-            .files(sources_to_compile);
+        // Handle compilation based on whether dual compilation is enabled
+        let (output, _dual_compiled_contracts) = if config.resolc.resolc_compile {
+            // Dual compilation mode: compile both solc and resolc
 
-        // Apply revive size limits if revive compilation is enabled
-        if config.resolc.resolc_compile {
-            compiler =
-                compiler.size_limits(revive::CONTRACT_SIZE_LIMIT, revive::CONTRACT_SIZE_LIMIT);
-        }
+            // Compile with solc to a subdirectory
+            let mut solc_config = config.clone();
+            solc_config.out = solc_config.out.join(revive::SOLC_ARTIFACTS_SUBDIR);
```

Comment:
you are still compiling to `pvm` because you didn't unset the `config.resolc = Default::default` for `solc` specifically.